tests/testthat/test-pstr_at_product_level.R

```diff
@@ -4,3 +4,13 @@ test_that("outputs expected columns at product level", {
   out <- pstr_at_product_level(companies, scenarios)
   expect_named(out, pstr_cols_at_product_level())
 })
+
+test_that("the thresholds are in the range 0 to 1", {
+  pstr_arguments <- formals(pstr_at_product_level)
+
+  low_threshold <- eval(pstr_arguments$low_threshold)
+  high_threshold <- eval(pstr_arguments$high_threshold)
+
+  expect_true(low_threshold >= 0 & low_threshold <= 1)
+  expect_true(high_threshold >= 0 & high_threshold <= 1)
+})
```

Comment:
While in general it IS advisable to test every exported function, I lean towards violating that rule here and instead relying on the more comprehensive tests in the top-level `pstr()` function. This makes the code more maintainable because the lower level `xxtr_at_*_level()` functions are still quite experimental so we intentionally change them a lot. If we had tests for all of them it would be a nightmare to adapt all of them each time we change them -- so we rely on that if something is really wrong we'll know from the tests of `pstr()`.

With that in mind, the tests on this lower level function are an exception, currently reserved to something there is no way to test from the top level `pstr()`.